### PR TITLE
[TypeScript Support ] Added types on top of existing types

### PR DESCRIPTION
**Pull request by ljosberinn**
existing types taken from [here](https://github.com/storeon/storeon/blob/master/index.d.ts)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * @param {String[]} paths The keys of state object
+ *    that will be store in local storage
+ * @param {Config} config The config object
+ * @param {String} [config.key='storeon'] The default key
+ *    to use in local storage
+ */
+declare function persistState<State>(
+  paths?: string[],
+  config?: StoreonLocalStorage.Config
+): createStore.Module<State>;
```
Comment (hadeeb) [inline index.d.ts]: You can do something like
```js
import Storeon from "storeon";

declare function persistState<State>(
  paths?: string[],
  config?: StoreonLocalStorage.Config
):Storeon.Module<State>;
```

We don't have to duplicate the type definition

**Comment by polemius** [comment]
LGTM :+1: 
@ai What do you think? Can we merge it?

**Comment by ai** [comment]
LGTM (but I have 0 experience with TS)

**Comment by ljosberinn** [comment]
FWIW I'm currently using it whilst developing an app with a modified 

> node_modules/@storeon/localstorage

 folder by adding the committed file, no issues so far.

**Comment by polemius** [comment]
@ljosberinn Thank you :+1:

**Pull request merged**